### ISSUE TITLE
Exception on set text and button is not click

### ISSUE THREAD

**Issue by crfnunes** [Roemer/FlaUI#116]
Hello,

I'm can't to use FlaUi, need help please.

First, I try set a text to a TextBox and showme a "System.InvalidOperationException".
```
var pass = window.FindFirstDescendant(cf => cf.ByName("ds_pass"))?.AsTextBox();
pass.Text = "SUPER";
```

Second, I try click a button, but not happens:

`window.FindFirstDescendant(cf => cf.ByName("OK"))?.AsButton().Click();`

Thanks!!

**Comment by danhi** [comment]
Try breaking this up into two statements:
var ae = window.FindFirstDescendant(cf => cf.ByName("ds_pass"));
Then check to make sure this isn't null before trying:
var pass = ae.AsTextBox;
Also, how are you setting window?

**Comment by crfnunes** [comment]
Hi,

Verify and variable "ae" is not null, but exception still occurred.
My code as below:

```
var app = Application.Launch(@"C:\Portal\portal.exe");
var automation = new UIA3Automation();
var window = app.GetMainWindow(automation);

var ae = window.FindFirstDescendant(cf => cf.ByName("ds_pass"));

if (ae != null)
{
      var pass= ae.AsTextBox();                   
      pass.Text = "SUPER";
}
```

* Button clicking now :).

```
var bt = window.FindFirstDescendant(cf => cf.ByName("OK"));             
bt.AsButton().Click();
```

Thank you!

**Comment by crfnunes** [comment]
Complement: 
Verify "pass" and is not null too.

**Comment by crfnunes** [comment]
Full exception:
System.InvalidOperationException 
  HResult=0x80131509
  Message=A operação não é válida devido ao estado atual do objeto.
  Source=Interop.UIAutomationClient
  StackTrace:
   em Interop.UIAutomationClient.IUIAutomationValuePattern.SetValue(String val)
   em FlaUI.UIA3.Patterns.ValuePattern.<>c__DisplayClass4_0.<SetValue>b__0()
   em FlaUI.Core.Tools.Com.Call(Action nativeAction)
   em FlaUI.UIA3.Patterns.ValuePattern.SetValue(String value)
   em FlaUI.Core.AutomationElements.TextBox.set_Text(String value)
   em Automation.FlaUI._1_TestFlaUI() em C:\Automation\FlaUI.cs:linha 34

**Comment by Roemer** [comment]
What kind of control is your textbox (WPF, WinForms, some library control)?
As alternative, you could try using `Enter` instead of `.Text`.

**Comment by Roemer** [comment]
Ah and another possibility: Is your TextBox disabled or set as read only?

**Comment by crfnunes** [comment]
Try with "Enter", but showme same exception and textbox is enable/editable. :(
The software is build on PowerBuilder.

**Comment by Roemer** [comment]
Can you post an "Inspect" screenshot of your TextBox showing the base properties and patterns (best is to use FlaUInspect)?

**Comment by Roemer** [comment]
Can you post an inspect screenshot of the control? Most prefferably with FlaUInspect.

**Comment by Roemer** [comment]
Closed due to inactivity. Please reopen if you have something to add.

**Comment by LevYas** [comment]
This question is searchable by Google and I come across it while working on my problem, hope it helps others.
I have WPF app, there's a textbox and button bound to a command. When you enter something into textbox, the button become enable. But, I had a flaky test related to it. It worked fine on local machine, but on CI I got timeout error while waiting for the button to enable:

System.TimeoutException : Timeout occurred in retry
  Stack Trace:
     at FlaUI.Core.Tools.Retry.While[T](Func`1 retryMethod, Func`2 checkMethod, Nullable`1 timeout, Nullable`1 interval, Boolean throwOnTimeout, Boolean ignoreException, String timeoutMessage, Boolean lastValueOnTimeout, T defaultOnTimeout)
   at FlaUI.Core.Tools.Retry.WhileFalse(Func`1 checkMethod, Nullable`1 timeout, Nullable`1 interval, Boolean throwOnTimeout, Boolean ignoreException, String timeoutMessage)
   at FlaUI.Core.AutomationElements.AutomationElementExtensions.WaitUntilEnabled[T](T self, Nullable`1 timeout)

I used textBox.Text to set the value, and apparently it didn't produce needed events. textBox.Enter() solved the problem, thank you Roemer!
